crates/database/src/app_data.rs

```diff
@@ -0,0 +1,83 @@
+use {crate::AppId, sqlx::PgConnection};
+
+/// Tries to associate the contract app data with the full app data.
+///
+/// If this contract app data already existed then the existing full app data is
+/// returned.
+///
+/// If this contract did not already exist, it is inserted and the passed in
+/// full app data is returned.
+pub async fn insert(
+    ex: &mut PgConnection,
+    contract_app_data: &AppId,
+    full_app_data: &[u8],
```

Comment:
Is it "contract app data", "app data hash", od "app id"? Would be nice to have one name for this concept instead of three, and always refer to it using that name.

Reply:
The type AppId should be renamed in this crate. I started using "contract app data" consistently in writing since working on the "revamp app data" issue.